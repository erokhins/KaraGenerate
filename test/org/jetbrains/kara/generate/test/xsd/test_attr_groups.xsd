<?xml version='1.0' encoding='utf-8' ?>
<xsd:schema xmlns='test' xmlns:xsd='http://www.w3.org/2001/XMLSchema' version="1.0" targetNamespace='test'>

    <xsd:attributeGroup name="group1">
        <xsd:attribute name="attrInGroup1"/>
    </xsd:attributeGroup>

    <xsd:attributeGroup name="group2">
        <xsd:attribute name="attrInGroup2"/>
    </xsd:attributeGroup>

    <xsd:attributeGroup name="group3">
        <xsd:attributeGroup ref="group1"/>
        <xsd:attributeGroup ref="group2"/>
        <xsd:attribute name="attrInGroup3"/>
    </xsd:attributeGroup>

    <xsd:element name="simple">
        <xsd:complexType>
            <xsd:attributeGroup ref="group3"/>
            <xsd:attribute name="attrInElement" />
        </xsd:complexType>
    </xsd:element>

</xsd:schema>