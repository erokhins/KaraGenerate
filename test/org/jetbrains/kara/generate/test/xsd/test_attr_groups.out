AttributeTypeDecl:
AttributeDecl:
   AttributeDecl = (attrInElement, null, null})
   AttributeTypeDecl = (string, null)

   AttributeDecl = (attrInGroup1, null, null})
   AttributeTypeDecl = (string, null)

   AttributeDecl = (attrInGroup2, null, null})
   AttributeTypeDecl = (string, null)

   AttributeDecl = (attrInGroup3, null, null})
   AttributeTypeDecl = (string, null)

AttributeGroupDecl:
   AttributeGroup = group1
   ParentGroups = []
   NewAttributes =
      AttributeDecl = (attrInGroup1, null, null})
      AttributeTypeDecl = (string, null)


   AttributeGroup = group2
   ParentGroups = []
   NewAttributes =
      AttributeDecl = (attrInGroup2, null, null})
      AttributeTypeDecl = (string, null)


   AttributeGroup = group3
   ParentGroups = [group1, group2]
   NewAttributes =
      AttributeDecl = (attrInGroup3, null, null})
      AttributeTypeDecl = (string, null)


ElementDecl:
   ElementDeclaration = (simple, false)
   ElementGroups = []
   NewAllowElements = []
   AttributeGroups = [group3]
   NewAttributes =
      AttributeDecl = (attrInElement, null, null})
      AttributeTypeDecl = (string, null)


ElementGroup:
